src/aurora/gateway/gateway.h

```diff
@@ -68,18 +71,40 @@ class Session : public std::enable_shared_from_this<Session> {
    * @param hostname The hostname to resolve.
    * @param port The port to use. If unsure, leave the default value.
    */
-  void Connect(std::string& hostname, const std::string& port = "443");
+  void Connect(std::string token, std::string& hostname,
+               const std::string& port = "443");
 
   /**
    * @brief Closes an active WebSocket connection.
    * @param code The close code to send to the server.
    */
   void Disconnect(const websocket::close_code& code);
 
+  /**
+   * @brief Start receiving certain gateway events
+   * @param intents Intent(s) category to subscribe to
+   */
+  void SubscribeTo(uint16_t intent);
```

Comment:
Since you're passing a bitmask consisting of a variable amount of intents here, consider renaming the parameter to `intents`.